inject-java/src/main/java/io/micronaut/annotation/processing/BeanDefinitionInjectProcessor.java

```diff
@@ -1258,6 +1258,11 @@ protected void accept(DeclaredType type, Element element, AopProxyWriter aopProx
                             int paramLen = targetParameters.size();
                             if (paramLen == sourceParams.length) {
```

Comment:
I was under the impression this check would return false for the example in the issue because the number of parameters in the method is different than the number of methods in the interface method

Reply:
Did some looking into. It appears the JavaMethodElement separates out the continuation argument thus getParameters() returns all other params

Reply:
Yup. I initially had a "fix" in a different location, then found the same thing when it wasn't hitting my change.